src/celeritas/track/SimTrackView.hh

```diff
@@ -255,10 +260,10 @@ CELER_FUNCTION void SimTrackView::force_step_limit(ActionId action)
  */
 CELER_FUNCTION void SimTrackView::force_step_limit(StepLimit const& sl)
 {
-    CELER_ASSERT(sl.step >= 0
-                 && sl.step <= states_.step_limit[track_slot_].step);
+    CELER_ASSERT(sl.step >= 0 && sl.step <= states_.steps[track_slot_]);
 
-    states_.step_limit[track_slot_] = sl;
+    states_.steps[track_slot_] = sl.step;
+    states_.post_step_action[track_slot_] = sl.action;
```

Comment:
We might be able to eliminate this function now that we can manipulate the length and action separately. It seems to be used only once in the main branch (DiscreteSelectExecutor) which was even eliminated during this update.